src/components/OneTimePopup/OneTimePopup.tsx

```diff
@@ -10,14 +10,18 @@ import Modal from '../dls/Modal/Modal';
 
 import styles from './OneTimePopup.module.scss';
 
-import { selectIsPopupVisible, setIsPopupVisible } from 'src/redux/slices/popup';
+import { selectSessionCount } from 'src/redux/slices/popup';
 import openGivingLoopPopup from 'src/utils/givingloop';
 
+const OPEN_POPUP_EVERY_N_SESSIONS = 10;
 const OneTimePopup = () => {
   const { t } = useTranslation('common');
-  const isPopupVisible = useSelector(selectIsPopupVisible);
+  const sessionCount = useSelector(selectSessionCount);
   const [isDonateButtonLoading, setIsDonateButtonLoading] = useState(false);
-  const dispatch = useDispatch();
+
+  const initialIsPopupVisible =
```

Comment:
nit: no need to have a var called `initialIsPopupVisible` we can simply calculate the value directly:

```
useState(()=> {
return sessionCount % OPEN_POPUP_EVERY_N_SESSIONS === 0 && sessionCount > 0
}}
```